### ISSUE TITLE
Send selected text from editor to console without prompts (docstrings)

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#114]
_From [david.fo...@gmail.com](https://code.google.com/u/107726689308966932341/) on 2009-11-10T09:19:33Z_

What steps will reproduce the problem?

 1. I have some docstrings in modules. I run my program in external consol
and I want to select and send these lines to the console to evaluate them
manually. Ex.

    """
    \>>> a=big_function_name( with, a lot,
              of, parameters, on, many lines)
    \>>> print(a)
    result
    """
2. I select from the beginning of a= to the end of the function and then I
press ctrl+f5 to send to console.

3. It should send only the selection, but instead sends things as if I had
selected the whole line including the >>> . So it makes an error. 

For now, I have to make ctrl+c ctrl+v in the console as a workaround. What is the expected output? What do you see instead? It should only send the selection, or there could be a filter to eliminate
any '>>> ' in the sent lines. It could be an option. What version of the product are you using? On what operating system? spyder 1.00  on ubuntu karmic 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=114_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-01-03T11:25:31Z_

There is already a similar option to copy text from the interactive console plugin
("Copy without prompts").
Could you please check this feature?
If it fit your needs, I suggest adding the same feature to the Editor plugin.

**Status:** Accepted

**Comment by spyder-issues-migrator** [comment]
_From [david.fo...@gmail.com](https://code.google.com/u/107726689308966932341/) on 2010-01-03T13:42:23Z_

The "Copy without prompts" is not a complete solution:

1) I believe that a selection of multiple lines, not starting at the beginning 
   of line should work. Here is a dumb example, where the underline means the 
   desired selection
 
var1,var2 = function( long_series, of, 
            ---------------------------
                       parameters, on, many, lines)
                       ----------------------------

  The current behavior of ctrl+5 will also send "var1,var2 = " to the console
  The desired behavior is to send only the underlined part, so it would print the 
  result ( in this exemple in a console)

2) The "Copy without prompts" does however takes care of the ">>>" for easy parsing  
   of docstring for one command. It is not quite correct for many commands with 
   indent:

   In command line I produce an indented prompt with print
\>>> print("""
...         >>> coord = s.array([[6, 6, 7, 7, 8, 8, 8, 8, 9, 9, 9, 9, 9],
...                            [1, 3, 4, 6, 7, 8, 9, 2, 3, 4, 9, 1, 2]])
...         >>> print(coord)
... """)

        \>>> Coord = s.array([[6, 6, 7, 7, 8, 8, 8, 8, 9, 9, 9, 9, 9],
                           [1, 3, 4, 6, 7, 8, 9, 2, 3, 4, 9, 1, 2]])
        \>>> print('%.4f, %.4f'  %baryCenter(Coord))

   If I Copy without prompts the above and paste, I get:

Coord = s.array([[6, 6, 7, 7, 8, 8, 8, 8, 9, 9, 9, 9, 9],
                           [1, 3, 4, 6, 7, 8, 9, 2, 3, 4, 9, 1, 2]])
        print(Coord)

   Which has an indentation error.

In conclusion, to ignore >>> in 'send to console' should probably be an option to
enable somewhere. It must work correctly with multiple line with indents. Besides,
the selected text of multiple lines should be send to console, with ctrl+F5, from the
beginning of the selection and not the beginning of the line.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-01-08T04:32:15Z_

With recent changes:
1) -> it now works

2) -> I think that this "send selected text to console without prompts (and with 
indentation fixed)" feature is getting too specialized. I'm tagging this Issue's 
priority as low.

**Summary:** Send selected text from editor to console without prompts (docstrings)  
  **Labels:** -Type-Defect -Priority-Medium Type-Enhancement Priority-Low

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2011-10-07T13:26:53Z_

**Labels:** Cat-Editor

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-10-29T04:16:13Z_

**Status:** WontFix
